### PR TITLE
Student: submit responses: allow student to add a comment for each response (Rubric and Msq questions) #7239

### LINKED ISSUE #7239: Student: submit responses: allow student to add a comment for each response
When students submit responses, it is useful if the system allows student to add an explanatory comment. It also avoid the need to add questions such as 'explain your response to the previous question' to collect justifications.

This might have to be done as multiple PRs, one per question type. We can start with MCQ questions.

### PR DESCRIPTION
**Pull request by Mynk96**
Fixes #7239

### CODE REVIEW

**Comment by Mynk96** [comment]
@chowyb Please Review!

**Comment by Mynk96** [comment]
@chowyb `verifyHtmlPart` will somewhat lead to verification of `main content` if there are many number of questions for a particular question types. For example, MCQ type.

**Comment by chowyb** [comment]
@Mynk96 true, but in the case of question types with fewer questions, this may still be a drawback.

Assuming we cover all question types during testing (and by extension, every question in the feedback session used for testing), I believe that the contents of each question should only need to be checked once per action (add without response, add with response, edit, delete). My concern is that ff using `verifyHtmlMainContent`, this could end up checking all the questions once per question type, resulting in checking up to 9 times, which may not be necessary.

**Comment by Mynk96** [comment]
@damithc your input?

**Comment by damithc** [comment]
> @damithc your input?

I'll have to defer to @chowyb 's judgement on this one :-) @chowyb please go ahead and suggest the approach you think the best for @Mynk96 to take.  I don't mind taking a slightly risky approach if there is a big benefit in terms of cost of testing.

**Comment by chowyb** [comment]
@Mynk96 after looking at the code again, I feel that perhaps the first option (doing full HTML comparison for tests by action type, rather than doing partial HTML comparison for tests by question type) would be better, due to the following:

1. Tests the full page instead of smaller sections of the page, therefore bugs changing other parts of the pages can be spotted.
2. Less HTML files required; although doing by question type and using `verifyHtmlPart` will result in the same amount of HTML comparisons, managing the large number of resource HTML files will likely prove problematic.

Again, there's still the problem of dependency for the first option, but this seems to be the better solution of the two in my opinion. A slight workaround would be to indicate that these test methods are dependent (eg. using `testEditCommentsAfterAdd` or similar as a method name).

**Comment by Mynk96** [comment]
@chowyb In that case, I'll have to change complete tests. Should I have a single PR to test all types of question?

**Comment by chowyb** [comment]
@Mynk96 it may be fine to just do both in a single PR then. From my understanding, allowing comments on every question type is relatively simple (overriding a function), and may be too small to necessitate having a PR on its own.

Note that even after these PRs get merged into `add-student-response-comment`, you will need to make one final PR to `master`, which will be mainly to fix any possible merge conflicts, as well as addressing review comments from other developers if any.

**Comment by Mynk96** [comment]
@chowyb Please Review!

**Comment by Mynk96** [comment]
@chowyb Please Review!

**Comment by Mynk96** [comment]
@chowyb Merge conflicts resolved.Please Review!

File: src/test/resources/pages/instructorFeedbackSubmitPageEditedComments.html~upstream_add-student-response-comment [hunk 1]
```diff
@@ -0,0 +1,4177 @@
+<div class="container" id="mainContent">
```
Comment (chowyb) [inline src/test/resources/pages/instructorFeedbackSubmitPageEditedComments.html~upstream_add-student-response-comment]: Remove this file.

Comment (Mynk96) [inline src/test/resources/pages/instructorFeedbackSubmitPageEditedComments.html~upstream_add-student-response-comment]: Removed!

**Comment by Mynk96** [comment]
@chowyb Ready For Review!

**Pull request merged**